packages/manager/src/utilities/filterPlanSelectionsByType.ts

```diff
@@ -0,0 +1,58 @@
+import { PlanSelectionType } from 'src/features/linodes/LinodesCreate/SelectPlanPanel';
+
+interface FilteredPlansTypes {
+  nanodes: PlanSelectionType[];
+  standard: PlanSelectionType[];
+  highMem: PlanSelectionType[];
+  proDedicated: PlanSelectionType[];
+  dedicated: PlanSelectionType[];
+  gpu: PlanSelectionType[];
+  metal: PlanSelectionType[];
+  premium: PlanSelectionType[];
+}
```

Comment:
This type could be simplifed to 
```suggestion
type FilteredPlansTypes = Record<LinodeTypeClass, PlanSelectionType[]>;
```

if you are willing to change `nanodes` to `nanode` and `highMem` to `highmem` and `prodDedicated` to `prodedicated`.

I think this change is optional, but I I find that matching the API's types as much as possible help keep things simple and more understandable